services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/modules/directory_watcher/_core.py

```diff
@@ -126,8 +127,9 @@ def set_enabled(self, is_enabled: bool) -> None:
         self._is_enabled = is_enabled
 
     def _invoke_push_directory(self) -> None:
-        # wrapping the function call in the object
-        # helps with testing, it is simplet to mock
+        if not self._is_enabled:
+            return
```

Comment:
very minor: not sure I get the point of moving this if from the ```on_any_event``` it was nice over there no?

Reply:
I did a lot of tries and at some point this function was called from different handlers. I left it here, where I think it should be located.